src/ty/mod.rs

```diff
@@ -247,14 +247,14 @@ pub enum TypeDef<T: Form = MetaForm> {
     Array(TypeDefArray<T>),
     /// A tuple type.
     Tuple(TypeDefTuple<T>),
-    /// A Range type.
-    Range(TypeDefRange<T>),
     /// A Rust primitive type.
     Primitive(TypeDefPrimitive),
     /// A type using the [`Compact`] encoding
     Compact(TypeDefCompact<T>),
     /// A type representing a sequence of bits.
     BitSequence(TypeDefBitSequence<T>),
+    /// A Range type.
+    Range(TypeDefRange<T>),
```

Comment:
Oh good point. Do you mind mentioning this in the docs as well?

Reply:
That it is backwards compatible, or that new variants should be added to the end?

Reply:
The latter. Something like "Tooling relying on scale-info data generally relies on order to find items. For this reason it is important that public enums add new variants items to the end to ensure backwards compatibility"?

Reply:
+1

Reply:
I've gone a step further and added explicit indices for this and `TypeDefPrimitive`, in case people delete or reorder variants.